DB/src/main/java/io/deephaven/db/tables/utils/TableManagementTools.java

```diff
@@ -45,40 +48,111 @@ private TableManagementTools() {
+        return getTable("Stand-alone V2 table from " + path, tableDefinition, locationProvider);
+    }
+
+    private static Table readTableFromSingleParquetFile(@NotNull final File sourceFile, @NotNull final TableDefinition tableDefinition) {
+        final TableLocationProvider locationProvider = new ReadOnlyLocalTableLocationProviderByParquetFile(
+                StandaloneTableKey.getInstance(),
+                sourceFile,
+                false,
+                TableDataRefreshService.getSharedRefreshService());
+        return getTable("Read single parquet file from " + sourceFile, tableDefinition, locationProvider);
     }
 
     /**
      * Reads in a table from disk.
      *
-     * @param path table location
+     * @param sourceFilePath table location
+     * @return table
+     */
+    public static Table readTable(@NotNull final String sourceFilePath) {
+        return readTable(new File(sourceFilePath));
+    }
+
+    /**
+     * Reads in a table from disk.
+     *
+     * @param sourceFile table location
+     * @return table
+     */
+    public static Table readTable(@NotNull final File sourceFile) {
+        return readParquetTable(sourceFile, false);
+    }
+
+    /**
+     * Reads in a table from disk.
+     *
+     * @param sourceDir table location
      * @return table
      */
-    public static Table readTable(@NotNull final File path) {
-        final TableDefinition tableDefinition = TableDefinition.loadDefinition(path, path.getName());
-        if (tableDefinition == null) {
-            throw new IllegalStateException(path + " doesn't have a loadable TableDefinition");
+    public static Table readTableFromDir(@NotNull final File sourceDir) {
+        return readParquetTable(sourceDir, true);
+    }
+
+    private static Table readParquetTable(@NotNull final File source, final boolean isDirectory) {
+        final ArrayList<ColumnDefinition> cols = new ArrayList<>();
+        final ParquetReaderUtil.ColumnDefinitionConsumer colConsumer =
+                (final String name, final Class<?> dbType, Class<?> componentType, final boolean isGrouping, final String codecName, final String codecArgs) -> {
+                    final int columnType = isGrouping ? ColumnDefinition.COLUMNTYPE_GROUPING : ColumnDefinition.COLUMNTYPE_NORMAL;
+                    final ColumnDefinition<?> colDef;
+                    if (codecName != null) {
+                        colDef = ColumnDefinition.ofVariableWidthCodec(name, dbType, columnType, componentType, codecName, codecArgs);
+                    } else {
+                        colDef = ColumnDefinition.fromGenericType(name, dbType, columnType, componentType);
+                    }
+                    cols.add(colDef);
+                };
+        try {
+            final String path = source.getPath() + ((!isDirectory) ? "" : File.separator + ParquetTableWriter.PARQUET_FILE_NAME);
+            ParquetReaderUtil.readParquetSchema(path, colConsumer);
```

Comment:
Because ParquetFileReader constructor wants a String.  So there is no point to do a fandango with a File here to get the string out of the File object later.